### ISSUE TITLE
Connection delegation does not pick the correct become_password

### ISSUE THREAD

**Issue by dw** [dw/mitogen#401]
via IRC:

````
17:10:42 <Overhacked> Hi there! Working on adopting ansible_mitogen into an existing project that admins FreeBSD jails. The Ansible-native jail plugin is limited to running locally on the FreeBSD jailhost. Is that also a limitation of the mitogen_jail plugin, or do I need to better understand how to configure ansible_mitogen to run on the controller?
17:40:49 <dw> Overhacked: nope, you can use it with mitogen_via=
17:41:15 <dw> Overhacked: define the jail parent host in inventory, then define the jail itself with connection=mitogen_jail mitogen_via=<parent-hostname>
17:41:35 <Overhacked> I've been playing with that and having some trouble with become
17:41:36 <dw> be careful to ensure your task limits don't accidentaly include the parent host :)
17:41:50 <dw> feel free to pastebin any issues
17:42:06 <Overhacked> Should the host be ansible_connection=mitogen_sudo
17:42:41 <dw> is your connection   <controller> -> ssh -> sudo -> jail ?
17:42:58 <Overhacked> Hmm, looking at config_from_hostvars, I see that it sets become_pass=None. I've been testing this all with ad-hoc commands. Let me try with a short playbook.
17:43:08 <dw> you can define a become user either explicitly using mitogen_sudo (which is mostly just for testing, don't recommend that), or using  mitogen_via=root@<jailparent>
17:43:28 <dw> ah, you may have found an untested case :S
17:52:11 <Overhacked> OK, let me try it using mitogen_via=USER@...
17:52:22 <Overhacked> FYI, here's the config I'm testing: https://gist.github.com/overhacked/7ba147b550b6d37e3a0b1a988af3315d
17:56:30 <Overhacked> Just updated the gist. Now with mitogen_via=root@jailhost, I get "sudo password is required" when the jail command tries to run.````

**Comment by dw** [comment]
@overhacked I've opened a bug here to track this. Should get to it soon

**Comment by dw** [comment]
See also https://github.com/dw/mitogen/issues/340#event-1815592830

**Comment by overhacked** [comment]
Hi there, @dw! Thanks for jumping on this. I dug a little more in after our conversation, but it quickly got deeper than I was prepared to go in the context of my current project. I do, however, think mitogen is pretty awesome, and I'm going to try to work it back into this project as time permits. It takes _forever_ for the playbook to run over SSH.

**Comment by dw** [comment]
Hopefully this and every bug like it will be squished before 0.2.4 is out. It requires a big rewrite of some code, though, so it may take until December as I'm no longer full time on the project

**Comment by dw** [comment]
Hi @overhacked, sorry for the crazy delay. I have confirmed this appears to be fixed as part of the 'holistic' work done in #412.

This should now be fixed on the master branch and will make it into the next release. To be updated when a new release is made, subscribe to https://www.freelists.org/list/mitogen-announce

If you are still experiencing the problem on the current master branch, please do not hesitate to reopen this issue.

Thanks for reporting this!
